kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java

```diff
@@ -447,4 +601,42 @@ public Object loadPlanningProblem(int computerListSize,
         }
         return problem;
     }
+
+    private ProblemFactChange loadAddProblemFactChange(final int currentNumberOfComputers) {
+        ProblemFactChange addProblemFactChange = null;
+        try {
+            Class<?> cbgc = kieContainer.getClassLoader().loadClass(CLASS_CLOUD_GENERATOR);
+            Object cbgi = cbgc.newInstance();
+
+            Method method = cbgc.getMethod("createCloudComputer",
+                                           int.class);
+            Object cloudComputer = method.invoke(cbgi,
+                                                 currentNumberOfComputers);
+
+            Class<?> cloudComputerClass = kieContainer.getClassLoader().loadClass(CLASS_CLOUD_COMPUTER);
+            Class<?> problemFactChangeClass = kieContainer.getClassLoader().loadClass(CLASS_ADD_COMPUTER_PROBLEM_FACT_CHANGE);
+
+            Constructor constructor = problemFactChangeClass.getConstructor(cloudComputerClass);
+            addProblemFactChange = (ProblemFactChange) constructor.newInstance(cloudComputer);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
@ge0ffrey +1, I'll rework this & other generators to catch specific exception types.